src/ncp/posix/netif.cpp

```diff
@@ -85,6 +87,38 @@ void Netif::Deinit(void)
     Clear();
 }
 
+void Netif::UpdateIp6UnicastAddresses(const std::vector<otIp6AddressInfo> &aAddresses)
+{
+    mIp6UnicastAddressesSwap.clear();
+    for (const otIp6AddressInfo &address : aAddresses)
+    {
+        mIp6UnicastAddressesSwap.emplace_back(address);
+    }
+
+    // Remove stale addresses
+    for (const Ip6AddressInfo &address : mIp6UnicastAddresses)
```

Comment:
avoid swap array with iterator?

Reply:
Thanks! Updated the implementation and removed the swap.